src/sgame/sg_bot_ai.cpp

```diff
@@ -1289,7 +1226,7 @@ AINodeStatus_t BotActionBuy( gentity_t *self, AIGenericNode_t *node )
 	if ( buy->nparams == 0 )
 	{
 		// equip action
-		BotGetDesiredBuy( self, &weapon, upgrades, &numUpgrades );
+		numUpgrades = BotGetDesiredBuy( self, weapon, upgrades, 4 );
```

Comment:
one less magic number here:
```suggestion
		numUpgrades = BotGetDesiredBuy( self, weapon, upgrades, ARRAY_LENGTH( upgrades ) );
```